### ISSUE TITLE
Share to Rocket.Chat is not working on iOS

### ISSUE THREAD

**Issue by bingluen** [RocketChat/Rocket.Chat.ReactNative#3932]
### Description:

Share to Rocket.Chat is not working on iOS

### Environment Information:

- Rocket.Chat Server Version: irrelevant
- Rocket.Chat App Version:  fc9e9a4f
- Device Name:  iPhone 12 simulator 
- OS Version: 15.2

### Steps to reproduce:

1. open safari
2.  go any website
3. click share button
4. select rocket.chat
### Expected behavior:

Should show dialog  can select workspace or channel  like android.

### Actual behavior:

https://user-images.githubusercontent.com/6832868/158949454-71a4f290-cf19-4e80-9d01-a0ec0d1be9f0.mp4

**Comment by diegolmello** [comment]
There's lots of possibilities.
Can you reproduce it from App Store?
Otherwise, you'll need to select `ShareRocketChatRN` scheme on Xcode and build it from there.

**Comment by bingluen** [comment]
> There's lots of possibilities.
> Can you reproduce it from App Store?
> Otherwise, you'll need to select `ShareRocketChatRN` scheme on Xcode and build it from there.

The version from App Store is working correctly.

Is `select ShareRocketChatRN scheme on Xcode` means following setting ?

I build previous version `857394ab` (4.23) in same way, but not occur this issue.

<img width="938" alt="image" src="https://user-images.githubusercontent.com/6832868/159198186-03c8ea41-b2e4-4658-897f-9acd2c4978fa.png">

I did not get any error message from metro console and Xcode log during execution share function

**Comment by GleidsonDaniel** [comment]
Are you using Intel or M1 macbook?

**Comment by bingluen** [comment]
> 

M1

**Comment by GleidsonDaniel** [comment]
Okay, it's a problem that's mapped out and we're going to work on a solution to that.
It works on Intel macbooks, but not on M1.

As it is a specific fix for macbooks M1 (Both Intel or physical devices works normally) it ended up going unnoticed.

**Comment by bingluen** [comment]
> Okay, it's a problem that's mapped out and we're going to work on a solution to that. It works on Intel macbooks, but not on M1.
> 
> As it is a specific fix for macbooks M1 (Both Intel or physical devices works normally) it ended up going unnoticed.

Is there any breaking change after 857394ab (4.23) ? 

This issue not only simulator on Apple silicon but also on physical device. (My device is iPhone XS Max with iOS 15.3.1)

I build 4.23 in same environment(Apple Silicon M1 Pro with iPhone XS Max), but not  occur this issue.

My repo (base on 857394ab) https://github.com/PDIS/Rocket.Chat.ReactNative/tree/4.23-pdis-rc

**Comment by bingluen** [comment]
I try to building with Intel machine, and get same issue.

**Comment by ArnavKaushik** [comment]
@bingluen Not sure if you're still facing this issue but I've noticed this issue a couple of days ago on iOS and found that the issue was related to the app groups that the share extension was assigned to (under Signings & Capabilities).

I also had an issue with no servers found once I got the extension to appear, but that was fixed by importing the `server` key from `app.json` and using it in the `init` function of `app/share.tsx` instead of the UserPreferences value of currentServer. This would only work if you are running the app through the single-server branch, as this wouldn't correctly assign the server for multiple server setups.
